Source/Scene/Cesium3DTileset.js

```diff
@@ -157,26 +161,29 @@ define([
         var url = options.url;
 
         //>>includeStart('debug', pragmas.debug);
-        Check.typeOf.string('options.url', url);
+        Check.defined('options.url', url);
         //>>includeEnd('debug');
 
-        var tilesetUrl;
+        var resource = Resource.createIfNeeded(url);
+
+        var tilesetResource = resource;
         var basePath;
 
-        if (getExtensionFromUri(url) === 'json') {
-            tilesetUrl = url;
+        if (resource.extension === 'json') {
             basePath = getBaseUri(url, true);
```

Comment:
`getBaseUri(url, true);` -> `getBaseUri(resource.url, true);`

Reply:
And the `isDataUri` call below it.  Perhaps both of these should be rolled into `Resource` anyway?